WordPress/src/main/java/org/wordpress/android/ui/qrcodeauth/QRCodeAuthViewModel.kt

```diff
@@ -12,25 +14,106 @@ import org.wordpress.android.ui.posts.BasicDialogViewModel.DialogInteraction.Neg
 import org.wordpress.android.ui.posts.BasicDialogViewModel.DialogInteraction.Positive
 import org.wordpress.android.ui.qrcodeauth.QRCodeAuthActionEvent.FinishActivity
 import org.wordpress.android.ui.qrcodeauth.QRCodeAuthActionEvent.LaunchDismissDialog
+import org.wordpress.android.ui.qrcodeauth.QRCodeAuthActionEvent.LaunchScanner
 import org.wordpress.android.ui.qrcodeauth.QRCodeAuthDialogModel.ShowDismissDialog
+import org.wordpress.android.ui.qrcodeauth.QRCodeAuthUiState.Loading
+import org.wordpress.android.util.NetworkUtilsWrapper
 import javax.inject.Inject
 
+@Suppress("TooManyFunctions")
 @HiltViewModel
-class QRCodeAuthViewModel @Inject constructor() : ViewModel() {
+class QRCodeAuthViewModel @Inject constructor(
+    private val uiStateMapper: QRCodeAuthUiStateMapper,
+    private val networkUtilsWrapper: NetworkUtilsWrapper,
+    private val validator: QRCodeAuthValidator
+) : ViewModel() {
     private val _actionEvents = Channel<QRCodeAuthActionEvent>(Channel.BUFFERED)
     val actionEvents = _actionEvents.receiveAsFlow()
 
+    private val _uiState = MutableStateFlow<QRCodeAuthUiState>(Loading)
+    val uiState: StateFlow<QRCodeAuthUiState> = _uiState
+
+    private var data: String? = null
+    private var token: String? = null
     private var isStarted = false
 
     fun start() {
         if (isStarted) return
         isStarted = true
+
+        updateUiStateAndLaunchScanner()
+    }
+
+    //  https://apps.wordpress.com/get/?campaign=login-qr-code#qr-code-login?token=asdfadsfa&data=asdfasdf
+    fun onScanSuccess(scannedValue: String?) {
+        handleScan(scannedValue)
+    }
+
+    fun onScanFailure() {
+        // Note: This is a result of the tap on "X" within the scanner view
+        postActionEvent(FinishActivity)
     }
 
     fun onBackPressed() {
         postActionEvent(LaunchDismissDialog(ShowDismissDialog))
     }
 
+    private fun cancelClicked() {
```

Comment:
Np: Can we rename `cancelClicked()` to `onCancelClicked()`, `authenticateClicked()` to `onAuthenticateClicked()` for consistency with other methods like: `onBackPressed()`, `onScanFailure()`, `onScanFailure()`?

Reply:
@ashiagr - great idea. Added in [fb04c17](https://github.com/wordpress-mobile/WordPress-Android/pull/16710/commits/fb04c1731c8ea836e4ef2add662d3fdbc259c714)